lib/Dialect/HW/HWOps.cpp

```diff
@@ -1053,40 +1054,26 @@ static void printModuleOp(OpAsmPrinter &p, ModuleTy mod) {
   }
 
   // Print the parameter list if present.
-  printOptionalParameterList(
-      p, mod.getOperation(),
-      mod.getOperation()->template getAttrOfType<ArrayAttr>("parameters"));
+  printOptionalParameterList(p, mod.getOperation(), mod.getParameters());
 
-  bool needArgNamesAttr = false;
-  module_like_impl::printModuleSignature(p, mod.getOperation(), argTypes,
-                                         /*isVariadic=*/false, resultTypes,
-                                         needArgNamesAttr);
+  module_like_impl::printModuleSignatureNew(p, mod.getOperation());
 
   SmallVector<StringRef, 3> omittedAttrs;
   if (isa<HWModuleGeneratedOp>(mod.getOperation()))
     omittedAttrs.push_back("generatorKind");
-  if (!needArgNamesAttr)
-    omittedAttrs.push_back("argNames");
-  omittedAttrs.push_back("portLocs");
-  omittedAttrs.push_back(
-      ModuleTy::getModuleTypeAttrName(mod.getOperation()->getName()));
-  omittedAttrs.push_back("per_port_attrs");
-  omittedAttrs.push_back("resultNames");
-  omittedAttrs.push_back("parameters");
+  omittedAttrs.push_back(mod.getPortLocsAttrName());
+  omittedAttrs.push_back(mod.getModuleTypeAttrName());
+  omittedAttrs.push_back(mod.getPerPortAttrsAttrName());
+  omittedAttrs.push_back(mod.getParametersAttrName());
   omittedAttrs.push_back(visibilityAttrName);
-  omittedAttrs.push_back(SymbolTable::getSymbolAttrName());
-  if (mod.getOperation()
-          ->template getAttrOfType<StringAttr>("comment")
-          .getValue()
-          .empty())
-    omittedAttrs.push_back("comment");
-  // inject argNames
-  auto attrs = mod->getAttrs();
-  SmallVector<NamedAttribute> realAttrs(attrs.begin(), attrs.end());
-  realAttrs.push_back(
-      NamedAttribute(StringAttr::get(mod.getContext(), "argNames"),
-                     ArrayAttr::get(mod.getContext(), mod.getInputNames())));
-  p.printOptionalAttrDictWithKeyword(realAttrs, omittedAttrs);
+  if (auto cmt =
+          mod.getOperation()->template getAttrOfType<StringAttr>("comment"))
+    if (cmt.getValue().empty())
+      omittedAttrs.push_back("comment");
```

Comment:
Consider using an explicit type instead of `auto` to avoid the `template`.

Reply:
That doesn't solve the function resolution issue.